### DISCUSSION TITLE
What is the role of train_size, test_data, and fold_strategy in the setup function?

### DISCUSSION THREAD

**Discussion by dpatchigolla** [pycaret/pycaret#3976] [Q&A]
In the docs, it says  test_data overrides train_size, and they are used as hold-out, and is used for validation. 

There's also fold_strategy param with stratified-k-fold as the default. So it appears we have train-cv, val-cv, and test_data - three datasets. How is test_data/train_size being used? In an older [discussion](https://github.com/pycaret/pycaret/discussions/3362#discussioncomment-5249572), it is mentioned that test_data is used only if 
cross_validation is set to False. So if I am using the default of cross_validation = True, and I don't pass a test_data, does my train_data not get split as per train_size param? 

Basically, are these three params mutually exclusive with fold_strategy > test_data > train_size as its priority order? Or is there a scenario where both fold_strategy and train_size (or test_data) are used in training/selecting a best model?
